### ISSUE TITLE
@segment/analytics-react-native-plugin-advertising-id declares minSdkVersion that prevents builds from working

### ISSUE THREAD

**Issue by tehlordvortex** [segmentio/analytics-react-native#626]
- `analytics-react-native` version: ^2.5.1
- Integrations versions (if used): `advertising-id` plugin, version ^1.0.1
- React Native version: Expo SDK 46, RN 0.69.3
- Android

**Steps to reproduce**
- Install `@segment/analytics-react-native-plugin-advertising-id`
- Use `expo-build-properties` to set `compileSdkVersion` to 31 as mentioned in the README
- Try to build the app

**Expected behavior**
- App builds successfully

**Actual behavior**
- Build fails with the following message:
```
[RUN_GRADLEW] /private/var/folders/_b/vjcx6rwd4nncrckh30193cmm0000gn/T/eas-build-local-nodejs/42a6346b-379f-46ec-b71f-a0f461b9df2a/build/android/app/src/debug/AndroidManifest.xml Error:
[RUN_GRADLEW]   uses-sdk:minSdkVersion 21 cannot be smaller than version 31 declared in library [:segment_analytics-react-native-plugin-advertising-id] /private/var/folders/_b/vjcx6rwd4nncrckh30193cmm0000gn/T/eas-build-local-nodejs/42a6346b-379f-46ec-b71f-a0f461b9df2a/build/node_modules/@segment/analytics-react-native-plugin-advertising-id/android/build/intermediates/merged_manifest/debug/AndroidManifest.xml as the library might be using APIs not available in 21
[RUN_GRADLEW]   Suggestion: use a compatible library with a minSdk of at most 21,
[RUN_GRADLEW]           or increase this project's minSdk version to at least 31,
[RUN_GRADLEW]           or use tools:overrideLibrary="com.analyticsreactnativepluginadvertisingid" to force usage (may lead to runtime failures)
[RUN_GRADLEW] FAILURE: Build failed with an exception.
```
This implies I have to set my `minSdkVersion` to 31, meaning my app will only be available for Android 12 and above.
But the Google Admobs documentation says `minSdkVersion` can be as low as 19:
<img width="560" alt="image" src="https://user-images.githubusercontent.com/21070796/183080138-928e4eca-fe5c-49e2-ba90-3a5598c68f71.png">

I believe the issue here is the configuration in the plugin's `build.gradle` is incorrect.

**Comment by jedashford** [comment]
PR to fix this here. https://github.com/segmentio/analytics-react-native/pull/630

**Comment by jedashford** [comment]
For those who are unable to build with this package, we used [patch-package ](https://www.npmjs.com/package/patch-package)to work around this for local and CI/CD.

```
diff --git a/node_modules/@segment/analytics-react-native-plugin-advertising-id/android/build.gradle b/node_modules/@segment/analytics-react-native-plugin-advertising-id/android/build.gradle
index 00e1318..8c973d2 100644
--- a/node_modules/@segment/analytics-react-native-plugin-advertising-id/android/build.gradle
+++ b/node_modules/@segment/analytics-react-native-plugin-advertising-id/android/build.gradle
@@ -28,7 +28,7 @@ def getExtOrIntegerDefault(name) {
 android {
   compileSdkVersion getExtOrIntegerDefault('compileSdkVersion')
   defaultConfig {
-    minSdkVersion 31
+    minSdkVersion 19
     targetSdkVersion getExtOrIntegerDefault('targetSdkVersion')
     versionCode 1
     versionName "1.0"

**Comment by alanjcharles** [comment]
fixed in [`1.0.2`](https://www.npmjs.com/package/@segment/analytics-react-native-plugin-advertising-id)
